### DISCUSSION TITLE
Welcome to MaterialX Discussions!

### DISCUSSION THREAD

**Discussion by bernardkwok** [autodesk-forks/MaterialX#1302] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by testpassword** [comment]
Dear Colleagues,

I am trying to start `JsMaterialXView` with my materials. I built the project with the command `npm run build` - the resulting files work fine with materials from the current repository, but when trying to open your mine material which has dependencies (files from folders `alglib`,` stdlib`, `bxdf`),` JsMaterialXGenShader.js` crashes with an error:

```
Error: Cycle detected!
http://localhost:8080/bxdf/disney_brdf_2012.mtlx
-> http://localhost:8080/bxdf/disney_brdf_2015.mtlx
-> http://localhost:8080/stdlib/stdlib_ng.mtlx
-> http://localhost:8080/stdlib/stdlib_defs.mtlx
-> http://localhost:8080/bxdf/substanceMetallicRoughness.mtlx
-> http://localhost:8080/bxdf/standard_surface.mtlx
-> http://localhost:8080/bxdf/ubershader_defs.mtlx
-> http://localhost:8080/bxdf/usd_preview_surface.mtlx
-> http://localhost:8080/alglib/alglib_defs.mtlx
-> http://localhost:8080/stdlib/stdlib_ng.mtlx
    promise http://localhost:8080/JsMaterialXGenShader.js:9
    promise callback*loadFiles http://localhost:8080/JsMaterialXGenShader.js:9
    promise http://localhost:8080/JsMaterialXGenShader.js:9
    promise callback*loadFiles http://localhost:8080/JsMaterialXGenShader.js:9
    _readFromXmlString http://localhost:8080/JsMaterialXGenShader.js:9
    readFromXmlString http://localhost:8080/JsMaterialXGenShader.js:9
    init webpack://jsmaterialxview/./src/index.js?:156
    promise callback*init webpack://jsmaterialxview/./src/index.js?:145
    <anonymous> webpack://jsmaterialxview/./src/index.js?:38
    js http://localhost:8080/main.js:149
    __webpack_require__ http://localhost:8080/main.js:173
    <anonymous> http://localhost:8080/main.js:213
    <anonymous> http://localhost:8080/main.js:215
index.js:241:17
    init webpack://jsmaterialxview/./src/index.js?:241
    (asynchronous: promise callback)
    init webpack://jsmaterialxview/./src/index.js?:240
    <anonymous> webpack://jsmaterialxview/./src/index.js?:38
    js http://localhost:8080/main.js:149
    __webpack_require__ http://localhost:8080/main.js:173
    <anonymous> http://localhost:8080/main.js:213
    <anonymous> http://localhost:8080/main.js:215
```

The main material (file `BrickWall_Red_Weathered.mtlx`) I put in the folder` resources/Materials/Examples/StandartSurface`, and dependencies (`alglib`,` bxdf`, `stdlib`,` textures`) to the build root after building `source/JsMaterialX/JsMaterialXView/dist`.

You can find the archive with the material and the full log of the browser with the error in [Google Drive](https://drive.google.com/drive/folders/1K5Lh_qMEeiB19Qy4PbPG0HAzeNeTjAw8?usp=sharing).

When viewing materials in the OS-native version of viewer, there is no problem throws, everything works correctly. I would like to know, am I doing something wrong, or are materials with dependencies not supported at the moment?

Thanks for the help

**Comment by bernardkwok** [comment]
Hi @testpassword (Artemy).
If you don't mind I will follow up on this in a new discussion thread (https://github.com/autodesk-forks/MaterialX/discussions/1312).
